src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,59 +1,36 @@
-package seedu.addressbook.data.person;
-
-import seedu.addressbook.data.exception.IllegalValueException;
-
-/**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
- */
-public class Address {
-
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = trimmedAddress;
-    }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
-}
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ */
+public class Address extends Contact {
+
+    EXAMPLE = "123, some street";
+    MESSAGE_CONSTRAINTS = "Person addresses can be in any format";
+    VALIDATION_REGEX = ".+" ;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public Address(String address, boolean isPrivate) throws IllegalValueException {
+        String trimmedAddress = address.trim();
+        this.isPrivate = isPrivate;
+        if (!isValid(trimmedAddress)) {

```

Comment:
The `isValid` method refers to the method in `Contact` class.
``` java
	    /**
 	     * Checks if a given string is a valid person phone number.
 	     */
 	    public static boolean isValid(String test) {
 	        return test.matches(VALIDATION_REGEX);
 	    }
```
Note that you cannot override static fields (e.g. `VALIDATION_REGEX`) in subclass.